### DISCUSSION TITLE
Remove "try another way" link

### DISCUSSION THREAD

**Discussion by emmarichardson** [sventorben/keycloak-home-idp-discovery#127] [General]
I have been trying to set this up and am probably just setting it wrong.  I need to redirect one domain to Google and the rest of the users to just normal username/password login.  I have it working as I need by using your example set up but because I have the IDP as an alternative, I get the Try Another Way link.  If I make IDP required, the non-Google users cannot login because they do not get the password entry page...would appreciate any assistance you can give me...

**Comment by sventorben** [comment]
Hello @emmarichardson, 
can I see a screenshot of your current flow configuration, please.

**Reply by emmarichardson** [reply]
<img width="1263" alt="Screen Shot 2022-12-22 at 6 20 42 AM" src="https://user-images.githubusercontent.com/2079129/209143089-9b0d382b-a8b8-419a-a4f5-ccc3e8f072ca.png">

Currently, I am just hiding the link with CSS but having issues with caching to get that to pull the correct css without disabling caching altogether!

**Reply by sventorben** [reply]
Ok, I need to check if there is another way to hide the link, but I am afraid there is not.

**Comment by emmarichardson** [comment]
I feel like there is something wrong though.  Ideally I should be able to make the IDP discovery required (which would hide the link) but it is not correctly forwarding to a password form when I do that - is that a bug or do I have it set up wrong?

**Reply by sventorben** [reply]
It's not possible to do so. If you offer a password form as an alternative way to login, then you obviously have two alternatives 1) discovery and 2) username/password.

**Comment by emmarichardson** [comment]
I was just thinking that IDP discovery should include the password form as the “ELSE” variant...

**Reply by sventorben** [reply]
How would that be different from what you have right now?

**Comment by emmarichardson** [comment]
You would only need the IDP option and would not need to add the password form because it would already be part of the IDP option.  Not a huge deal, hopefully I can figure out the caching issues and the css link hide will work.

**Reply by sventorben** [reply]
But then you would always have a password form and are unable to configure flows that want passwordless logins.
